reStream.sh

```diff
@@ -86,6 +79,65 @@ if ! ssh_cmd true; then
     exit 1
 fi
 
+rm_version="$(ssh_cmd cat /sys/devices/soc0/machine)"
+
+# We cannot read the framebuffer directory in the reMarkable 2, but we know the
+# address of the pointer to the framebuffer for some reMarkable 2 versions.
+# This will need to be changed each time xochitl updates.
+rm2_getpointer() {
+    xochitl_version="$(ssh_cmd "sed -n 's/^REMARKABLE_RELEASE_VERSION=//p' /usr/share/remarkable/update.conf")"
+    case "$xochitl_version" in
+        "2.3.*")
+            echo "4121024"
+            ;;
+        "2.4.*")
+            # TODO
+            echo "???"
+            ;;
+        *) # use last known version if we cannot find a match
+            echo "WARNING: unknown reMarkable 2 release version" >&2
+            echo "consider updating if reStream doesn't work" >&2
+            echo "???"
+            ;;
+    esac
+}
+
+case "$rm_version" in
+    "reMarkable 1.0")
+        width=1408
+        height=1872
+        bytes_per_pixel=2
+        pixel_format="rgb565le"
+        # calculate how much bytes the window is
+        window_bytes="$((width * height * bytes_per_pixel))"
+        # read the first $window_bytes of the framebuffer
+        head_fb0="dd if=/dev/fb0 count=1 bs=$window_bytes 2>/dev/null"
+        ;;
+    "reMarkable 2.0")
+        width=1404
+        height=1872
+        bytes_per_pixel=1
+        pixel_format="gray8"
+        pid="$(ssh_cmd pidof xochitl)"
+        pointer="$(rm2_getpointer)"
+        read_address="addr=\$(dd if=/proc/$pid/mem bs=1 count=4 skip=$pointer 2>/dev/null | hexdump | awk '{print \$3\$2\}') && printf '%d' \$((16#\$addr))"
```

Comment:
There's an extra `\` here.
```suggestion
        read_address="addr=\$(dd if=/proc/$pid/mem bs=1 count=4 skip=$pointer 2>/dev/null | hexdump | awk '{print \$3\$2}') && printf '%d' \$((16#\$addr))"
```